hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonQueryFunction.java

```diff
@@ -32,22 +32,24 @@
 import com.hazelcast.sql.impl.type.QueryDataType;
 import org.apache.calcite.sql.SqlJsonQueryEmptyOrErrorBehavior;
 import org.apache.calcite.sql.SqlJsonQueryWrapperBehavior;
-import org.jsfr.json.exception.JsonPathCompilerException;
 import org.jsfr.json.path.JsonPath;
 
 import java.io.IOException;
+import java.io.ObjectInputStream;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Objects;
+import java.util.function.Function;
 
 import static com.hazelcast.jet.sql.impl.expression.json.JsonPathUtil.serialize;
 import static com.hazelcast.jet.sql.impl.expression.json.JsonPathUtil.wrapToArray;
 
 @SuppressWarnings("checkstyle:MagicNumber")
 public class JsonQueryFunction extends VariExpression<HazelcastJsonValue> implements IdentifiedDataSerializable {
     private static final ILogger LOGGER = Logger.getLogger(JsonQueryFunction.class);
+    private static final Function<String, JsonPath> COMPILE_FUNCTION = JsonPathUtil::compile;
 
-    private final Cache<String, JsonPath> pathCache = JsonPathUtil.makePathCache();
+    private transient ConcurrentInitialSetCache<String, JsonPath> pathCache = JsonPathUtil.makePathCache();
```

Comment:
We can cache a constant path here in the same way as in `JsonValueFunction`

Reply:
Yeah, when I started my work on that I didn't know what is the difference :)